### PR TITLE
Support package names other than directory base path name

### PR DESCRIPTION
**Pull request by dvic**
This PR changes how the output package name is chosen. Instead of using the base path, it first checks if there is a go file in the working directory and uses that package name. In this way, the generated file will contain the correct package name if it is different than the base path (which is allowed in Go).

### CODE REVIEW

**Comment by dvic** [comment]
By the way, I think https://github.com/vektah/gqlgen also has the same issue, I guess we should also fix it there?

**Comment by vektah** [comment]
Thanks for this, yeah theres an open issue for the gqlgen side but its going to be a bit more complicated because there are some circular data requirements https://github.com/vektah/gqlgen/issues/114.

File: dataloaden.go [hunk 1]
```diff
@@ -63,9 +65,12 @@ func getData(typeName string) (templateData, error) {
 		return templateData{}, fmt.Errorf("cant determine working dir: %s", err.Error())
 	}
 
-	pkgName := strings.Join(parts[:len(parts)-1], ".")
+	pkgData, err := getPackage(wd)
+	if err != nil {
+		return templateData{}, fmt.Errorf("failed to resolve output package name: %s", err.Error())
```
Comment (vektah) [inline dataloaden.go]: could this fall back to the old behavior instead?

Comment (dvic) [reply dataloaden.go]: The fallback is done in `getPackage`, but not if `parser.ParseDir` returns an error (not sure when this exactly happens with `parser.PackageClauseOnly`). Should I ignore this error?

Comment (vektah) [reply dataloaden.go]: yeah handling it by falling back to the directory name seems ok.

Comment (dvic) [reply dataloaden.go]: Done, I've also rebased the commit and updated the example with the changes from the previous PR.

**Pull request merged**